core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java

```diff
@@ -141,7 +141,7 @@ private void adjustRuntimeImage(JReleaserContext context, JpackageAssembler asse
 
         try {
             if (!FileUtils.copyFilesRecursive(context.getLogger(), originalImage, adjustedImage, path -> {
-                boolean pathIsJar = path.getFileName().toString().endsWith(".jar") && path.getParent().getFileName().toString().equals("jars");
+                boolean pathIsJar = path.getFileName().toString().endsWith(".jar") && "jars".equals(path.getParent().getFileName().toString());
```

Comment:
This conditional could still throw NPEs if an intermediate result in the chain `path.getParent().getFileName().toString()` were to be null. However, it won't. I think this change may be omitted.

Reply:
```suggestion
                boolean pathIsJar = path.getFileName().toString().endsWith(".jar") && path.getParent().getFileName().toString().equals("jars");
```

Reply:
Thought I could do this quickly from GitHub, but forgot about the preference for squashed & commit message conventions; I'll force push! :)

Reply:
It's OK. I've squashed on my side 😏